### PR TITLE
Int: FIxing integer conversion problems

### PR DESCRIPTION
**Pull request by otaviobp**

### CODE REVIEW

File: src/lib/int.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#ifndef SWUPD_INT_H
+#define SWUPD_INT_H
+
+/**
+ * @file
+ * @brief Basic string operations.
```
Comment (castulo) [inline src/lib/int.h:6]: you forgot to update the @brief

File: src/swupd_lib/signature.c [hunk 2]
```diff
@@ -186,8 +186,19 @@ bool signature_verify_data(const void *data, size_t data_len, const void *sig_da
 	BIO *verify_BIO = NULL;
 	char *errorstr = NULL;
 	PKCS7 *p7 = NULL;
+	int sig_data_len_int, data_len_int;
 
-	sig_BIO = BIO_new_mem_buf(sig_data, sig_data_len);
+	if (ulong_to_int_err(sig_data_len, &sig_data_len_int) != 0) {
+		error("Data to big to be a signature file (size = %ld)", sig_data_len);
```
Comment (castulo) [inline src/swupd_lib/signature.c:192]: missing \n

File: src/swupd_lib/signature.c [hunk 3]
```diff
@@ -186,8 +186,19 @@ bool signature_verify_data(const void *data, size_t data_len, const void *sig_da
...
+		goto error;
+	}
+
+	if (ulong_to_int_err(data_len, &data_len_int) != 0) {
+		error("Data to big to verify signature (size = %ld)", data_len);
```
Comment (castulo) [inline src/swupd_lib/signature.c:197]: same here

**Comment by otaviobp** [comment]
V2: Fixing reported problems

File: src/lib/int.c [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+#include "macros.h"
+
+#include <errno.h>
+#include <limits.h>
+
+int long_to_ulong_err(long a, unsigned long *b)
+{
+	if (!b) {
+		UNEXPECTED();
+		debug("Invalid parameter: b is NULL");
```
Comment (castulo) [inline src/lib/int.c:12]: I just noticed you also forgot the \n in all these debug messages 😓

**Comment by otaviobp** [comment]
V3: Adding \n on debug messages

**Pull request merged**